### DISCUSSION TITLE
批量测试 Cloudflare CDN 第三方镜像 IP

### DISCUSSION THREAD

**Discussion by e5u** [XIU2/CloudflareSpeedTest#192] [分享]
最近发现了 [一个仓库项目](https://github.com/ip-scanner/cloudflare)，里面提供了 Cloudflare CDN 的第三方镜像 IP。
将仓库文件 [ZIP 打包下载](https://github.com/ip-scanner/cloudflare/archive/refs/heads/daily.zip)并解压后，使用下面的 .bat 脚本将所有 .txt 文件内容合并及加上 /32 后缀，
最后就可以使用 ` -f ` 参数指定刚刚生成的 `all.txt` 进行测速了。

```
IP 地址           已发送  已接收  丢包率  平均延迟  下载速度 (MB/s)
112.29.7.86       4       4       0.00    32.39     16.23
117.161.143.162   4       4       0.00    29.28     3.14
112.28.199.105    4       4       0.00    30.97     1.42
117.161.143.157   4       4       0.00    29.28     0.38
116.136.163.162   4       4       0.00    30.95     0.32
112.29.7.84       4       4       0.00    32.80     0.29
36.102.227.162    4       4       0.00    33.02     0.11
211.137.139.60    4       4       0.00    32.42     0.00
111.19.167.95     4       4       0.00    33.54     0.00
110.17.168.196    4       4       0.00    33.64     0.00
```

``` bat
@echo off

type *.txt>>all.txt
for /f "tokens=*" %%i in (all.txt) do echo %%i/32>>$
move $ all.txt
```

**Comment by mingxiaoyu** [comment]
有意思

**Comment by garygao1965** [comment]
Thanks!

**Comment by denniszzg** [comment]
感谢。
请教一下，这个CF的第三方镜像IP是什么意思？使用起来有无安全隐患？比如暴露后面的VPS地址，暴露访问纪录等等

**Reply by XIU2** [reply]
第三方指的就是非官方了。
即个人搭建的 Cloudflare CDN 反代（中转/镜像），一般都是它们自用的，但是因为没有做安全措施，导致被别人扫描出来了（用套了 CF 的域名指向该 IP 访问，能访问就说明是 CF CDN 反代，其实用我的 CloudflareST 的下载测速也是这个原理，只不过用来扫描的话效率不高。。

正常访问的网络过程：
```
访客 <=> Cloudflare CDN <=> 源站
```
使用这些反代后的网络过程：
```
访客 <=> 该反代 <=> Cloudflare CDN <=> 源站
```

并无你提到的安全隐患，但是这些反代都是个人搭建自用的，一旦用的人多了，就会被发现并关闭，因此没啥稳定性可言。

**Reply by denniszzg** [reply]
非常感谢科普。
当时第一感觉是，担心会否有钓鱼的可能。当我的v2ray客户端填写了这些第三方IP，我访问外网时我的隐私信息就被纪录下来了。

**Reply by denniszzg** [reply]
听说，有些VPN服务商有政府背景，意味着使用该VPN就会暴露隐私。
同理，会否有官方背景建立CF CDN反代，可以通过分析追溯到使用者？
![1](https://user-images.githubusercontent.com/30626214/189264878-189b08b6-00b4-4d33-be3d-16c5b18c64a4.png)

**Reply by 954532838** [reply]
> 听说，有些VPN服务商有政府背景，意味着使用该VPN就会暴露隐私。 同理，会否有官方背景建立CF CDN反代，可以通过分析追溯到使用者？ ![1](https://user-images.githubusercontent.com/30626214/189264878-189b08b6-00b4-4d33-be3d-16c5b18c64a4.png)
我也担心，但是套上全局tls应该没事吧
